spec.html

```diff
@@ -4926,6 +5100,90 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
       </emu-note>
     </emu-clause>
 
+    <emu-clause id="sec-newprivatename" aoid="NewPrivateName">
+      <h1>NewPrivateName ( _description_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_description_) is String.
+        1. Return a new unique Private Name whose [[Description]] value is _description_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldfind" aoid="PrivateFieldFind">
+      <h1>PrivateFieldFind ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: _O_ is an object with a [[PrivateFieldValues]] internal slot.
+        1. For each element _entry_ in _O_.[[PrivateFieldValues]],
+          1. If _entry_.[[PrivateName]] is _P_, return _entry_.
+        1. Return ~empty~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldadd" aoid="PrivateFieldAdd">
+      <h1>PrivateFieldAdd ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be ! PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
```

Comment:
is there a reason that PrivateFieldFind returns empty, instead of throwing the TypeError, since every usage of it throws the same error on empty?

Reply:
That's not what happens: https://github.com/tc39/ecma262/pull/1655#discussion_r311954295

Reply:
oops, thanks, missed that reply.